hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Hangman.Game do
+
+  defstruct(
+    secret:      [],
+    word_chars:  MapSet.new(),
+    used_chars:  MapSet.new(),
+    game_state:  :initializing,
+    turns_left:  7,
+    letters:     [],
+    last_guess:  ""
+  )
+
+  def new_game() do
+    secret  = Dictionary.random_word() |> String.codepoints
+    %Hangman.Game{
+      secret:     secret,
+      word_chars: MapSet.new(secret),
+      letters:    String.duplicate("_", length(secret)) |> String.codepoints
+    }
+  end
+
+  def tally(game) do
+    %{
+      game_state: game.game_state,
+      turns_left: game.turns_left,
+      letters:    game.letters,
+      used:       game.used_chars |> MapSet.to_list(),
+      last_guess: game.last_guess
+    }
+  end
+
+  #               won?   old?   good?  turns_left
```

Comment:
This is an interesting approach: basically what in the old days was called a decision table.

+3 for laying it out in columns :)